### PR TITLE
Task complite.

### PR DESCRIPTION
**Pull request by DanilaSamusev**

### CODE REVIEW

File: Module2_4/Module2_4/Application.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+                    Console.WriteLine(result);
+                    break;
+                }
+            }
+        }
+
+        static Shape ChooseShape()
+        {
+            Console.Write("Inter the name of shape: circle, triangle or square - ");
```
Comment (az00s) [inline Module2_4/Module2_4/Application.cs]: Enter

File: Module2_4/Module2_4/Application.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            }
+
+            return requiredShape;
+        }
+
+        static string ChooseShapeData()
+        {
+            Console.Write("Inter perimeter or area - ");
+            string consoleData = Console.ReadLine();
+            string shapeData = null;
```
Comment (az00s) [inline Module2_4/Module2_4/Application.cs]: return consoleData == "area" || consoleData == "perimeter" ? consoleData  : null;

File: Module2_4/Module2_4/ShapeManager.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Module2_4
+{
+    public class ShapeManager
+    {
+        public double AskParameter(Shape shape)
+        {
+            double shapeParameter = 0;
+            
+            if (shape is Triangle)
```
Comment (az00s) [inline Module2_4/Module2_4/ShapeManager.cs]: switch is better here because a lot of if's

File: Module2_4/Module2_4/StringParser.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace Module2_4
+{
+    public class StringParser
```
Comment (az00s) [inline Module2_4/Module2_4/StringParser.cs]: use just DataParser for example - StringParser means you will get string as result

File: Module2_4/Module2_4/Entities/Triangle.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        public double SideLength { get; set; }
+        
+        public override void InitializeByParameter(double parameter)
+        {
+            SideLength = parameter;
+            Perimeter = SideLength * 3;
+            Area = Math.Pow(SideLength, 2) * Math.Sqrt(3) / 4;
+        }
+
+        public override void InitializeByPerimeter(double perimeter)
```
Comment (az00s) [inline Module2_4/Module2_4/Entities/Triangle.cs]: GetPerimeter

File: Module2_4/Module2_4/Entities/Triangle.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            InitializeByParameter(perimeter / 3);
+        }
+
+        public override void InitializeByArea(double area)
```
Comment (az00s) [inline Module2_4/Module2_4/Entities/Triangle.cs]: GetArea

File: Module2_4/Module2_4/Entities/Triangle.cs (hunk 5 above)
Comment (DanilaSamusev) [reply Module2_4/Module2_4/Entities/Triangle.cs]: I guess Get method returns  smth, but it doesn't

Comment (az00s) [reply Module2_4/Module2_4/Entities/Triangle.cs]: Ok, You're right.

**Pull request merged**